モデルのテスト

drumnaoto opened the pull request:
# WHAT
モデルのテスト実施。

# WHY
動作保証の為。
![2017-11-09 14 41 51](https://user-images.githubusercontent.com/33175652/32590241-c3756664-c55c-11e7-9373-acac2b0d9e7d.png)

drumnaoto committed:
add-gem-for-test

drumnaoto committed:
add-test-helper

追加したgemに記述を追加しました。

drumnaoto committed:
edit-validation-error

エラー名の
「name」を「グループ名」にしていると、ユーザー登録の際の「name」も「グループ名」になってしまうため、バリデーションエラーの表示の仕方を変
えました。
また、画像登録のエラーが続いたため、carrierwaveエラーの日本語化を行いました。
原因はImagemagikにありました。

drumnaoto committed:
add-validation-at-usersmodel

カラム作成時には一意制制約をかけていたが、
モデル自体に一意制制約をかけていなかったため、バリデーションを設定しました。

drumnaoto committed:
cat

テスト用の猫画像を追加しました。

drumnaoto committed:
test-at-user-model

userモデルでのテスト。

drumnaoto committed:
test-at-group-model

グループモデルでのテスト。

drumnaoto committed:
test-at-message-model

メッセージモデルでのテスト。

yamamotokeishiro commented on spec/factories/messages.rb:
```diff
@@ -0,0 +1,10 @@
+    body             "hello"
+    image            Rack::Test::UploadedFile.new(Rails.root.join("spec/fixtures/images/neko.jpg"))
+    user
+    group
```
アソシエーションgoodです！

yamamotokeishiro commented on spec/factories/messages.rb:
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+
+  factory :message do
+    body             "hello"
```
Fakerで生成しましょう！

yamamotokeishiro commented on spec/factories/groups.rb:
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+
+  factory :group do
+    name          "naotoman"
```
Fakerで生成しましょう！

yamamotokeishiro commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,10 @@
+  factory :user do
+    name                  "naoto"
+    email                 "naoto@gmail.com"
+    password              "00000000"
```
こちらもFakerを使用できます！

yamamotokeishiro commented:
設定周りをFakerを使用しましょう！

drumnaoto committed:
add-gem-faker

drumnaoto committed:
edit-name-by-faker

fakerを使って、groupのnameをランダムに作成しました。

drumnaoto committed:
edit-body-by-faker

fakerを使って、messageのbodyをランダムに作成しました。

drumnaoto committed:
edit-user-by-faker

fakerを使って、userのnameとemailをランダムに作成しました。
passwordはpassword_confirmationと同じ値にするためにfakerは使用していません。

drumnaoto committed:
edit-user_spec

fakerを使ったのでnameとemailがランダムに生成されるため、
重複した場合のvalidationをテストするために、同じ値を定義しました。

TOMARUMARU commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,10 @@
+  factory :user do
+    name                   { Faker::Name.name }
+    email                  { Faker::Internet.email }
+    password               "00000000"
```
passwordもFakerで作成してください。

drumnaoto committed:
make-fake_password

password用のfakeデータを作りました。

drumnaoto committed:
edit-user_spec

fakerを用いたpasswordのテストのため、
記述の変更をしました。
また、下の「7文字以上では登録不可」のテストの記述が、6文字しかなかったため、7文字に変更しました。

drumnaoto committed:
move-gem-file

rails sがfakerのname errorで起動しなかったため、
テスト用のGemをtest環境下に移動しました。
